##GATK: v4.2.2.0
import socket
sample_IDs, =glob_wildcards("path/to/input/tumor_bam/{sample}.tumor.bam")

tissue_types=['tumor','normal']
# a pseudo-rule that collects the target files
rule all:
	input:
		"path/to/output/normal_cohort.snps.VQSR.vcf.gz",
		"path/to/output/tumor_cohort.snps.VQSR.vcf.gz",
		"path/to/output/normal_cohort.indels.VQSR.vcf.gz",
		"path/to/output/tumor_cohort.indels.VQSR.vcf.gz"

rule HaplotypeCaller_single_tumor:
	input:
		fa="resources/Homo_sapiens_assembly38.fasta",
		bam="path/tpo/input/tumor_bam/{sample}.tumor.bam",
		interval="resources/S31285117_Padded.list"
	output:
		temp("path/to/output/vcfs_germline/{sample}.tumor.unsorted.g.vcf.gz")
	shell:
		"gatk --java-options \"-Xmx4g\" HaplotypeCaller -R {input.fa} \
		-I {input.bam} -ERC GVCF -L {input.interval} -O {output}"

rule HaplotypeCaller_single_normal:
	input:
		fa="resources/Homo_sapiens_assembly38.fasta",
		bam="path/to/input/normal_bam/{sample}.normal.bam",
		interval="resources/S31285117_Padded.list"
	output:
		temp("path/to/output/vcfs_germline/{sample}.normal.unsorted.g.vcf.gz")
	shell:
		"gatk --java-options \"-Xmx4g\" HaplotypeCaller -R {input.fa} \
		-I {input.bam} -ERC GVCF -L {input.interval} -O {output}"

rule sorted g_vcf:
	input:
		"path/to/output/vcfs_germline/{sample}.{type}.unsorted.g.vcf.gz"
	output:
		"path/to/output/vcfs_germline/{sample}.{type}.g.vcf.gz"
	shell:
		"gatk SortVcf I={input} O={output}"

rule list_all_vcfs:
	input:
		normal_vcfs=expand("path/to/output/vcfs_germline/{sample}.normal.g.vcf.gz",sample=sample_IDs),
		tumor_vcfs=expand("path/to/output/vcfs_germline/{sample}.tumor.g.vcf.gz",sample=sample_IDs),
	output:
		normal_list="path/to/output/normal.vcf.list",
		tumor_list="path/to/output/tumor.vcf.list"
	params:
		dir="path/to/output/vcfs_germline"
	shell:
		"find {params.dir}/ -name \"*.normal.g.vcf.gz\" > {output.normal_list};"
		"find {params.dir}/ -name \"*.tumor.g.vcf.gz\" > {output.tumor_list};"

rule CombineGVCFs_tumor:
	input:
		variant_list="path/to/output/tumor.vcf.list",
		reference="resources/Homo_sapiens_assembly38.fasta",
		interval="resources/S31285117_Padded.list"
	output:
		vcf="path/to/output/tumor.g.vcf.gz",
		index="path/to/output/tumor.g.vcf.gz.tbi"
	resources:
		mem_mb=120000
	shell:
		"gatk CombineGVCFs --java-options \"-Xmx120G -XX:ParallelGCThreads=8\" \
		-R {input.reference} -L {input.interval} --variant {input.variant_list} \
		-O {output.vcf}"

rule CombineGVCFs_normal:
	input:
		variant_list="path/to/output/normal.vcf.list",
		reference="resources/Homo_sapiens_assembly38.fasta",
		interval="resources/S31285117_Padded.list"
	output:
		vcf="path/to/output/normal.g.vcf.gz",
		index="path/to/output/normal.g.vcf.gz.tbi"
	resources:
		mem_mb=120000
	shell:
		"gatk CombineGVCFs --java-options \"-Xmx120G -XX:ParallelGCThreads=8\" \
		-R {input.reference} -L {input.interval} --variant {input.variant_list} \
		-O {output.vcf}"

rule GenotypeGVCFs_tumor:
	input:
		fa="resources/Homo_sapiens_assembly38.fasta",
		vcf="path/to/output/tumor.g.vcf.gz",
		index="path/to/output/tumor.g.vcf.gz.tbi"
	output:
		"path/to/output/all_tumor_samples.vcf.gz"
	shell:
		"gatk --java-options \"-Xmx20g\" GenotypeGVCFs -R {input.fa} \
		-V {input.vcf} -O {output}"

rule GenotypeGVCFs_normal:
	input:
		fa="resources/Homo_sapiens_assembly38.fasta",
		vcf="path/to/output/normal.g.vcf.gz",
		index="path/to/output/normal.g.vcf.gz.tbi"
	output:
		"path/to/output/all_normal_samples.vcf.gz"
	shell:
		"gatk --java-options \"-Xmx20g\" GenotypeGVCFs -R {input.fa} \
		-V {input.vcf} -O {output}"

rule VariantFiltration_tumor:
	input:
		"path/to/output/all_tumor_samples.vcf.gz"
	output:
		"path/to/output/all_tumor_samples_excesshet.vcf.gz"
	shell:
		"gatk --java-options \"-Xmx3g -Xms3g\" VariantFiltration -V {input} \
		--filter-expression \"ExcessHet > 54.69\" \
		--filter-name ExcessHet -O {output}"

rule VariantFiltration_normal:
	input:
		"path/to/output/all_normal_samples.vcf.gz"
	output:
		"path/to/output/all_normal_samples_excesshet.vcf.gz"
	shell:
		"gatk --java-options \"-Xmx20g -Xms20g\" VariantFiltration -V {input} \
		--filter-expression \"ExcessHet > 54.69\" \
		--filter-name ExcessHet -O {output}"

rule create_sites_only_vcf_normal:
	input:
		"path/to/output/all_normal_samples_excesshet.vcf.gz"
	output:
		"path/to/output/all_normal_samples_sitesonly.vcf.gz"
	shell:
		"gatk MakeSitesOnlyVcf -I {input} -O {output}"

rule create_sites_only_vcf_tumor:
	input:
		"path/to/output/all_tumor_samples_excesshet.vcf.gz"
	output:
		"path/to/output/all_tumor_samples_sitesonly.vcf.gz"
	shell:
		"gatk MakeSitesOnlyVcf -I {input} -O {output}"

rule VQSR_tumor_INDEL:
	input:
		vcf="path/to/output/all_tumor_samples_sitesonly.vcf.gz",
		resource1="resources/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz",
		resource2="resources/Axiom_Exome_Plus.genotypes.all_populations.poly.hg38.vcf.gz",
		resource3="resources/Homo_sapiens_assembly38.dbsnp138.vcf"
	output:
		tumor_indel_recal="path/to/output/tumor_cohort_indels.recal",
		tranches_file="path/to/output/tumor_cohort_indels.tranches"
	shell:
		"gatk --java-options \"-Xmx8g -Xms6g\" VariantRecalibrator -V {input.vcf} \
		--trust-all-polymorphic -tranche 100.0 -tranche 99.95 -tranche 99.9 \
		-tranche 99.5 -tranche 99.0 -tranche 97.0 -tranche 96.0 -tranche 95.0 \
		-tranche 94.0 -tranche 93.5 -tranche 93.0 -tranche 92.0 -tranche 91.0 -tranche 90.0 \
		-an FS -an ReadPosRankSum -an MQRankSum -an QD -an SOR -an DP -mode INDEL --max-gaussians 6 \
		--resource:mills,known=false,training=true,truth=true,prior=12 {input.resource1} \
		--resource:axiomPoly,known=false,training=true,truth=false,prior=10 {input.resource2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2 {input.resource3} \
		-O {output.tumor_indel_recal} --tranches-file {output.tranches_file}"

rule VQSR_tumor_SNP:
	input:
		vcf="path/to/output/all_tumor_samples_sitesonly.vcf.gz",
		resource1="resources/hapmap_3.3.hg38.vcf.gz",
		resource2="resources/1000G_omni2.5.hg38.vcf.gz",
		resource3="resources/1000G_phase1.snps.high_confidence.hg38.vcf.gz",
		resource4="resources/Homo_sapiens_assembly38.dbsnp138.vcf"
	output:
		tumor_SNP_recal="path/to/output/tumor_cohort_snps.recal",
		tranches_file="path/to/output/tumor_cohort_snps.tranches"
	shell:
		"gatk --java-options \"-Xmx3g -Xms3g\" VariantRecalibrator -V {input.vcf} \
		--trust-all-polymorphic -tranche 100.0 -tranche 99.95 -tranche 99.9 \
		-tranche 99.8 -tranche 99.6 -tranche 99.5 -tranche 99.4 -tranche 99.3 \
		-tranche 99.0 -tranche 98.0 -tranche 97.0 -tranche 90.0 -an QD -an MQRankSum \
		-an ReadPosRankSum -an FS -an MQ -an SOR -an DP -mode SNP --max-gaussians 6 \
		--resource:hapmap,known=false,training=true,truth=true,prior=15 {input.resource1} \
		--resource:omni,known=false,training=true,truth=true,prior=12 {input.resource2} \
		--resource:1000G,known=false,training=true,truth=false,prior=10 {input.resource3} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=7 {input.resource4} \
		-O {output.tumor_SNP_recal} --tranches-file {output.tranches_file}"

rule applyVQSR_tumor_INDEL:
	input:
		vcf="path/to/output/all_tumor_samples_sitesonly.vcf.gz",
		recal_file="path/to/output/tumor_cohort_indels.recal",
		tranches_file="path/to/output/tumor_cohort_indels.tranches",
	output:
		"path/to/output/tumor_cohort.indels.VQSR.vcf.gz"
	shell:
		"gatk --java-options \"-Xmx5g -Xms5g\" ApplyVQSR -V {input.vcf} \
		--recal-file {input.recal_file} --tranches-file {input.tranches_file} \
		--truth-sensitivity-filter-level 99.7 --create-output-variant-index true \
		-mode INDEL -O {output}"

rule applyVQSR_tumor_SNP:
	input:
		vcf="path/to/output/all_tumor_samples_sitesonly.vcf.gz",
		recal_file="path/to/output/tumor_cohort_snps.recal",
		tranches_file="path/to/output/tumor_cohort_snps.tranches",
	output:
		"path/to/output/tumor_cohort.snps.VQSR.vcf.gz"
	shell:
		"gatk --java-options \"-Xmx5g -Xms5g\" ApplyVQSR -V {input.vcf} \
		--recal-file {input.recal_file} --tranches-file {input.tranches_file} \
		--truth-sensitivity-filter-level 99.7 --create-output-variant-index true \
		-mode SNP -O {output}"

rule VQSR_normal_INDEL:
	input:
		vcf="path/to/output/all_normal_samples_sitesonly.vcf.gz",
		resource1="resources/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz",
		resource2="resources/Axiom_Exome_Plus.genotypes.all_populations.poly.hg38.vcf.gz",
		resource3="resources/Homo_sapiens_assembly38.dbsnp138.vcf"
	output:
		normal_indel_recal="path/to/output/normal_cohort_indels.recal",
		tranches_file="path/to/output/normal_cohort_indels.tranches"
	shell:
		"gatk --java-options \"-Xmx20g -Xms20g\" VariantRecalibrator -V {input.vcf} \
		--trust-all-polymorphic -tranche 100.0 -tranche 99.95 -tranche 99.9 \
		-tranche 99.5 -tranche 99.0 -tranche 97.0 -tranche 96.0 -tranche 95.0 \
		-tranche 94.0 -tranche 93.5 -tranche 93.0 -tranche 92.0 -tranche 91.0 -tranche 90.0 \
		-an FS -an ReadPosRankSum -an MQRankSum -an QD -an SOR -an DP -mode INDEL --max-gaussians 6 \
		--resource:mills,known=false,training=true,truth=true,prior=12 {input.resource1} \
		--resource:axiomPoly,known=false,training=true,truth=false,prior=10 {input.resource2} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=2 {input.resource3} \
		-O {output.normal_indel_recal} --tranches-file {output.tranches_file}"

rule VQSR_normal_SNP:
	input:
		vcf="path/to/output/all_normal_samples_sitesonly.vcf.gz",
		resource1="resources/hapmap_3.3.hg38.vcf.gz",
		resource2="resources/1000G_omni2.5.hg38.vcf.gz",
		resource3="resources/1000G_phase1.snps.high_confidence.hg38.vcf.gz",
		resource4="resources/Homo_sapiens_assembly38.dbsnp138.vcf"
	output:
		normal_SNP_recal="path/to/output/normal_cohort_snps.recal",
		tranches_file="path/to/output/normal_cohort_snps.tranches"
	shell:
		"gatk --java-options \"-Xmx3g -Xms3g\" VariantRecalibrator -V {input.vcf} \
		--trust-all-polymorphic -tranche 100.0 -tranche 99.95 -tranche 99.9 \
		-tranche 99.8 -tranche 99.6 -tranche 99.5 -tranche 99.4 -tranche 99.3 \
		-tranche 99.0 -tranche 98.0 -tranche 97.0 -tranche 90.0 -an QD -an MQRankSum \
		-an ReadPosRankSum -an FS -an MQ -an SOR -an DP -mode SNP --max-gaussians 6 \
		--resource:hapmap,known=false,training=true,truth=true,prior=15 {input.resource1} \
		--resource:omni,known=false,training=true,truth=true,prior=12 {input.resource2} \
		--resource:1000G,known=false,training=true,truth=false,prior=10 {input.resource3} \
		--resource:dbsnp,known=true,training=false,truth=false,prior=7 {input.resource4} \
		-O {output.normal_SNP_recal} --tranches-file {output.tranches_file}"

rule applyVQSR_normal_INDEL:
	input:
		vcf="path/to/output/all_normal_samples_sitesonly.vcf.gz",
		recal_file="path/to/output/normal_cohort_indels.recal",
		tranches_file="path/to/output/normal_cohort_indels.tranches",
	output:
		"path/to/output/normal_cohort.indels.VQSR.vcf.gz"
	shell:
		"gatk --java-options \"-Xmx5g -Xms5g\" ApplyVQSR -V {input.vcf} \
		--recal-file {input.recal_file} --tranches-file {input.tranches_file} \
		--truth-sensitivity-filter-level 99.7 --create-output-variant-index true \
		-mode INDEL -O {output}"

rule applyVQSR_normal_SNP:
	input:
		vcf="path/to/output/all_normal_samples_sitesonly.vcf.gz",
		recal_file="path/to/output/normal_cohort_snps.recal",
		tranches_file="path/to/output/normal_cohort_snps.tranches",
	output:
		"path/to/output/normal_cohort.snps.VQSR.vcf.gz"
	shell:
		"gatk --java-options \"-Xmx5g -Xms5g\" ApplyVQSR -V {input.vcf} \
		--recal-file {input.recal_file} --tranches-file {input.tranches_file} \
		--truth-sensitivity-filter-level 99.7 --create-output-variant-index true \
		-mode SNP -O {output}"
