
sample_IDs, =glob_wildcards("path/to/input/{sample}.tumor.bam")

rule all:
	input:
		expand("path/to/output/filter/{sample}.vcf.gz", sample=sample_IDs),

rule octopus:
	input:
		fa = "resources/Homo_sapiens_assembly38.fasta",
		tumor_bam = "path/to/input/{sample}.tumor.bam",
        normal_bam = "path/to/input/{sample}.normal.bam",
		region = "resources/S31285117_Padded_addchrM.list"
	output:
		out = "path/to/output/output/{sample}.vcf.gz"
	params:
		normal = "{sample}.normal"
	resources:
		mem_mb=12000
	shell:
		"""
		octopus -R {input.fa} -I {input.tumor_bam} {input.normal_bam} \
		-N {params.normal} -t {input.region} --threads 4 -o {output.out}
		"""

rule re_filter:
	input:
		fa = "resources/Homo_sapiens_assembly38.fasta",
		vcf = "path/to/output/output/{sample}.vcf.gz",
		somatic_forest = "resources/somatic.v0.7.4.forest",
		germline_forest = "resources/germline.v0.7.4.forest",
		tumor_bam = "path/to/output/{sample}.tumor.bam",
		normal_bam = "path/to/output/{sample}.normal.bam"
	output:
		out = "path/to/output/filter/{sample}.vcf.gz"
	params:
		normal = "{sample}.normal"
	shell:
		"""
		octopus -R {input.fa} -I {input.tumor_bam} {input.normal_bam} -N {params.normal}\
    			--filter-vcf {input.vcf} \
    			--forest {input.germline_forest} \
    			--somatic-forest {input.somatic_forest}\
    			-o {output.out}
		"""
