
sample_IDs, =glob_wildcards("path/to/input/{sample_id}.tumor.bam")

#localrules:all

# a pseudo-rule that collects the target files
rule all:
	input:
		expand("path/to/output_dir/results/variants/somatic.snvs.vcf.gz",sample=sample_IDs),
		expand("path/to/output_dir/results/variants/somatic.indels.vcf.gz",sample=sample_IDs)

rule StrelkaSomaticWorkflow:
	input:
		normal_bam="path/to/input/{sample}.normal.bam",
		tumor_bam="path/to/input/{sample}.tumor.bam",
		fa="resources/Homo_sapiens_assembly38.fasta",
		bed="resources/region.bed.gz"
	params:
		output_dir="demo/strelka_somatic/{sample}_strelka" #change as you you want
	output:
		"path/to/output_dir/runWorkflow.py"
	shell:
		"configureStrelkaSomaticWorkflow.py --normalBam {input.normal_bam} \
		--tumorBam {input.tumor_bam} --referenceFasta {input.fa} --exome \
		--callRegions {input.bed} --output_dir {params.output_dir}"

rule somatic_calling:
	input:
		"demo/strelka_somatic/{sample}_strelka/runWorkflow.py"
	output:
		snvs="path/to/output_dir/results/variants/somatic.snvs.vcf.gz",
		indels="path/to/output_dir/results/variants/somatic.indels.vcf.gz"
	shell:
		"path/to/output_dir/runWorkflow.py -m local -j 2"
